chord_metadata_service/chord/workflows/wdls/experiments_extended_json.wdl

```diff
@@ -0,0 +1,404 @@
+    }
+}
+
+task prepare_for_download {
+    input {
+        File json_document
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+with open('~{json_document}', 'r') as file:
+    data = json.load(file)
+
+download_list = [{'url': item['url'], 'filename': item['filename']} for item in data]
+
+print(json.dumps(download_list))
+    "
+    >>>
+    output {
+         Array[Map[String, String]] download_list = read_json(stdout())
+    }
+}
+
+task download_file {
+    input {
+        String url
+        String filename
+        String directory
+        String access_token
+        Boolean validate_ssl
+    }
+
+    command <<<
+    curl -L "~{url}" -o "~{filename}" -H "Authorization: Bearer ~{access_token}"
+    >>>
+
+    output {
+        File downloaded_file = '~{filename}'
+    }
+}
+
+task prepare_for_drs {
+    input {
+        File json_download_list
+        File json_path_list
+        Array[File]? downloaded_files
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+# Convert the space-separated string of downloaded files into a Python list
+downloaded_files = \"\"\"~{sep=' ' downloaded_files}\"\"\".split(' ')
+
+with open('~{json_download_list}', 'r') as file:
+    download_list = json.load(file)
+
+with open('~{json_path_list}', 'r') as file:
+    path_list = json.load(file)
+
+consolidated_paths = downloaded_files[:]
+
+consolidated_paths.extend([str(path['url']) for path in path_list])
+
+# Filter out any empty strings from consolidated_paths before printing
+filtered_paths = [path for path in consolidated_paths if path.strip()]
+
+print(json.dumps(filtered_paths))
+
+    "
+    >>>
+    output {
+        Array[String] consolidated_paths_for_drs = read_json(stdout())
+    }
+}
+
+task post_to_drs {
+    input {
+        File file_path
+        String drs_url
+        String project_dataset
+        String token
+        Boolean validate_ssl
+    }
+    command <<<
+        project_id=$(python3 -c 'print("~{project_dataset}".split(":")[0])')
+        dataset_id=$(python3 -c 'print("~{project_dataset}".split(":")[1])')
+        curl ~{true="" false="-k" validate_ssl} \
+            -X POST \
+            -F "file=@~{file_path}" \
+            -F "project_id=$project_id" \
+            -F "dataset_id=$dataset_id" \
+            -H "Authorization: Bearer ~{token}" \
+            --fail-with-body \
+            "~{drs_url}/ingest"
+    >>>
+    output {
+        String response_message = read_string(stdout())
+    }
+}
+
+task write_drs_responses_to_file {
+    input {
+        Array[String] drs_responses
+    }
+
+    command <<<
+        for response in "~{sep=' ' drs_responses}"; do
+            echo "$response"
+        done > results_post_drs.txt
+    >>>
+
+    output {
+        File results_post_drs = "results_post_drs.txt"
+    }
+}
+
+task parse_txt {
+    input {
+        File txt_responses
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+def read_and_process_file(file_path):
```

Comment:
What is the responsibility of this task?

Reply:
yes< I think a more descriptive name would be good.